src/Components/DynamicDrink.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+
+const DynamicDrink = (props) => {
+    const [ingredients, setIngredients] = useState([]);
+
+    useEffect(() => {
+        if (props.drink) {
+            const drinkInfo = props.drink[0];
+            // Making an array with all the ingredients & measurements
+            let ingredientArray = [];
+            for (let i = 1; i < 16; i++) {
+                const measurement = `strMeasure${i}`
+                const ingredient = `strIngredient${i}`
+                const ingredientObj = {
+                    [drinkInfo[ingredient]]: drinkInfo[measurement]
+                }
+                ingredientArray.push(ingredientObj)
+            }
+            // Making a new array with only the existing ingredients
+            const filteredIngList = ingredientArray.filter((ing) => {
+                const value = Object.values(ing)[0]
+                return ing[value] !== null && ing[value] !== ""
+            })
+
+            setIngredients(filteredIngList);
+        }
+    }, [props])
```

Comment:
Looks like an API call is made for every instance the form is changed by the user. You could also look into storing the data and calling it only once call amounts ever became an issue.